src/app/projects/components/project-settings/helper/data-handler-helper.ts

```diff
@@ -84,8 +87,14 @@ export class DataHandlerHelper {
         const values = settingModals.embedding.create.embeddingCreationFormGroup.getRawValue();
         let toReturn = this.getAttributeArrayAttribute(values.targetAttribute, 'name', attributes);
         toReturn += "-" + (values.granularity == 'ON_ATTRIBUTE' ? 'classification' : 'extraction');
-        toReturn += "-" + values.embeddingHandle;
-
+        const platform = settingModals.embedding.create.embeddingCreationFormGroup.get('platform').value
+        if (platform == PlatformType.HUGGING_FACE || platform == PlatformType.PYTHON) {
+            toReturn += "-" + values.model;
+        } else if (platform == PlatformType.OPEN_AI) {
+            toReturn += "-" + values.model + "-" + values.apiToken;
```

Comment:
i think the api token isn't (or at least shouldn't) be used in the final name. Not sure what the logic you and simon decided on was for the name but I think some kind of obfuscation or reduced values or something similar. Since names need to be unique this should be included here as well.